### PR TITLE
added __version__ to init to display version of the python package

### PR DESCRIPTION
**Pull request by kushagra219**
this is a part of my hacktoberfest2020 goals. Please review and see if it's okay.

### CODE REVIEW

File: s3_file_field/version.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+"""Version file"""
+
+__version__ = "0.0.13"
```
Comment (banesullivan) [inline s3_file_field/version.py:3]:
```suggestion
__version__ = "0.0.14"
```

File: s3_file_field/__init__.py [hunk 2]
```diff
@@ -4,3 +4,9 @@
 from .fields import S3FileField  # noqa: F401
 
 default_app_config = 's3_file_field.apps.S3FileFieldConfig'
+
+version = {}
+with open("s3_file_field/version.py") as fp:
+    exec(fp.read(), version)
+
+__version__ = version["__version__"]
```
Comment (banesullivan) [inline s3_file_field/__init__.py:12]: Why this approach and not?:

```py
from .version import __version__
```

----

Also, using `exec` in the package is frowned upon

File: setup.py [hunk 3]
```diff
@@ -156,10 +156,14 @@ def prerelease_local_scheme(version):
 with open('README.md') as f:
     readme = f.read()
 
+version = {}
+with open("s3_file_field/version.py") as fp:
+    exec(fp.read(), version)
+
```
Comment (banesullivan) [inline setup.py:162]: Okay to use `exec` here, I suppose

**Review by banesullivan — COMMENTED** [review]
thanks for addressing #104, @kushagra219

**Review by brianhelba — CHANGES_REQUESTED** [review]
Thanks for the contribution. I see a few significant issues so far:
* We use `setuptools-scm` to dynamically look up the version from the most recent Git tag. This means that hardcoded version numbers should not be committed anywhere in the codebase.
  * The use of `version='0.0.3'` is indeed a bug, and if you submit a PR just removing that, I'd accept it. :smile: 
* Using `exec` in almost any context is scary. I don't think it's worth using here.
  * I assume you're following (3) from [Single-sourcing the package version](https://packaging.python.org/guides/single-sourcing-package-version/). I suspect a fully working solution might require using `importlib.metadata` from (5).

**Comment by kushagra219** [comment]
@brianhelba I have tried to make the requested changes using importlib.metadata as suggested by you and removed exec.

**Comment by kushagra219** [comment]
@brianhelba @banesullivan @jbeezley @jeffbaumes sir, it would be a great help if you could just add the label of **`hacktoberfest-accepted`** to this PR.

File: setup.py [hunk 4]
```diff
@@ -156,10 +157,21 @@ def prerelease_local_scheme(version):
...
 
+
+def get_version(package):
+    """
+    Return package version as listed in `__version__` in `init.py`.
+    """
+    init_py = open(os.path.join(package, '__init__.py')).read()
+    return re.search("__version__ = ['\"]([^'\"]+)['\"]", init_py).group(1)
+
+version = get_version('s3_file_field')
```
Comment (brianhelba) [inline setup.py:168]: This entire code block will be non-functional. The `version` being passed here is ignored. The option `use_scm_version` overwrites the version dynamically.

**Review by brianhelba — CHANGES_REQUESTED** [review]
@kushagra219 Thanks for the updates. There are still a few issues with this PR, which I've left comments about inline.

**Comment by brianhelba** [comment]
@kushagra219 If you're looking for an easy PR to have accepted for Hacktoberfest, just update this PR to *only* remove the line
```python
version='0.0.3',
```
from `setup.py`. That will improve this code by fixing a source of confusion (since the `version` there is not really used), and since you found the bug, you deserve full credit for the fix.

I will happily approve and merge that change.

**Comment by kushagra219** [comment]
@brianhelba Sir, I have removed that version line from `setup.py` as you said and removed all other changes. Please see to it.

File: s3_file_field/__init__.py [hunk 5]
```diff
@@ -4,3 +4,13 @@
...
 
 default_app_config = 's3_file_field.apps.S3FileFieldConfig'
+
+try:
+    from importlib import metadata
+except ImportError:
+    # Running on pre-3.8 Python; use importlib-metadata package
+    import importlib_metadata as metadata
+
+assert metadata.version('django-s3-file-field') == '0.0.14'
```
[the remark this answers is not in the document]
Comment (kushagra219) [reply s3_file_field/__init__.py]: ok got it!

File: s3_file_field/__init__.py [hunk 6]
```diff
@@ -4,3 +4,13 @@
 from .fields import S3FileField  # noqa: F401
 
 default_app_config = 's3_file_field.apps.S3FileFieldConfig'
+
+try:
+    from importlib import metadata
+except ImportError:
+    # Running on pre-3.8 Python; use importlib-metadata package
+    import importlib_metadata as metadata
```
[the remark this answers is not in the document]
Comment (kushagra219) [reply s3_file_field/__init__.py]: It has come from the point (5) of [Single-sourcing the package version](https://packaging.python.org/guides/single-sourcing-package-version/) as you mentioned to do in the above comment.

**Review by brianhelba — APPROVED** [review]
Thanks for the improvement. I'd agree this helps remove confusion.

**Pull request merged**